absl/container/internal/common.h

```diff
@@ -55,7 +55,7 @@ class node_handle_base {
  public:
   using allocator_type = Alloc;
 
-  constexpr node_handle_base() {}
+  node_handle_base() = default;
```

Comment:
I think we want to keep `constexpr` on these, rather than let it be implicitly constexpr for the reasons you mentioned, but I'm unsure. As for the others, I'm also unsure, but @EricWF may have more information about whether this is okay.

Reply:
I agree that we want to keep the `constexpr`.

We should only omit `constexpr` when we want it to be deduced. When we require it, we should write it explicitly. Then the compiler will (hopefully) tell us right away if our expectation is violated.

Reply:
I would also agree, for me it is a deficiency in the standard, that `constexpr foo() = default;` is not possible. I will try to write a paper for that